### PR TITLE
DACT-1198 Directors' date of change - frontend.

### PR DESCRIPTION
**Pull request by mdesaikns**

### CODE REVIEW

File: src/controllers/update/update.director.check.answers.controller.ts [hunk 1]
```diff
@@ -62,6 +62,7 @@ const renderPage = async (req: Request, res: Response, companyNumber: string, of
     nationalityLink: urlUtils.getUrlToPath("", req),
     occupationLink: urlUtils.getUrlToPath("", req),
     dateUpdatedLink: urlUtils.getUrlToPath("", req),
-    correspondenceAddressChangeLink: urlUtils.getUrlToPath("", req)
+    correspondenceAddressChangeLink: urlUtils.getUrlToPath("", req),
+    currentUrl: req.originalUrl,
```
Comment (jditchburn) [inline src/controllers/update/update.director.check.answers.controller.ts:66]: What is this currentUrl parameter doing?

**Review by jditchburn — APPROVED** [review]

File: src/controllers/update/update.director.check.answers.controller.ts (hunk 1 above)
Comment (mdesaikns) [reply src/controllers/update/update.director.check.answers.controller.ts:66]: adds a language banner on the CYA page when `LOCALES_ENABLED` is enabled, without it I ran into the "Something went wrong" page.

File: locales/en/errors.json [hunk 2]
```diff
@@ -1,3 +1,14 @@
 {
-  "error-first-name-blank": "Enter the director’s full first name" 
+  "error-first-name-blank": "Enter the director’s full first name", 
+  "error-change-date-missing-day": "Date the director’s details changed must include a day",
+  "error-change-date-missing-month": "Date the director’s details changed must include a month",
+  "error-change-date-missing-year": "Date the director’s details changed must include a year",
+  "error-change-date-missing-day-month": "Date the director’s details changed must include a day and month",
+  "error-change-date-missing-day-year": "Date the director’s details changed must include a month and year",
+  "error-change-date-missing-day-month-year": "Enter the date the director’s details changed",
+  "error-change-date-invalid": "Date the director’s details changed must be a real date",
```
Comment (PhilipGlover) [inline locales/en/errors.json:9]: Some of the text seems a little unfamiliar to me.  If this is what UX / Analysis have provided, fair enough.  But comparing with the GDS guide `https://design-system.service.gov.uk/components/date-input/#error-messages`
Should this be more like "Date of change must be a real date" ?
and below "Date of change must be the same as or after [appointment date], when the director was appointed"
This is more to ask if the GDS guide has been referred to.  Wouldn't affect my approving.

**Review by PhilipGlover — APPROVED** [review]
lgtm
My comment re the wording of the validation messages may need to be considered outside of my approving this PR.

File: locales/en/errors.json (hunk 2 above)
Comment (mdesaikns) [reply locales/en/errors.json:9]: Hey @PhilipGlover.
I will highlight these concerns to our BA who will discuss this with the UX/relevant team. 
Although regarding "_Date of change must be the same as or after [appointment date], when the director was appointed_" will require injecting the dynamic input in the error message, unsure if we can expose an actual appointment date.

**Pull request merged**